packages/babel-plugin/src/utils/build-styled-component.ts

```diff
@@ -102,6 +101,60 @@ const traverseStyledBinaryExpression = (node: t.BinaryExpression, nestedVisitor:
   return node;
 };
 
+/**
+ * Returns a list of destructured props used in expressions of a styled component
+ *
+ * For example:
+ * ```
+ * const Component = styled.div`
+ *  width: $({ width }) => `${width}px`;
+ *  height: $({ height }) => `${height}px`;
+ * `;
+ * ```
+ * Returns list of [width, height]
+ *
+ * @param node {t.Node} Node of the styled component
+ */
+const getDestructuredProps = (node: t.Node): string[] => {
+  const destructuredProps: string[] = [];
+
+  const appendDestructuredPropsVisitor = {
+    noScope: true,
+    ObjectPattern(path: NodePath<t.ObjectPattern>) {
+      const { listKey, key } = path;
+
+      if (listKey === 'params' && key === 0) {
+        // Uses the destructure ObjectPattern path to get all the prop references, i.e.
+        // { width } becomes width
+        // { width, height } becomes width,height
+        // { size: { width } } becomes size:{width}
+        // { width: alias } becomes width:alias
+        const propsUsed = path.toString().replace(/\s/g, '').slice(1, -1).split(',');
+
+        destructuredProps.push(...propsUsed);
+        path.stop();
+      }
+    },
+  };
+
+  const arrowFunctionVisitor = {
+    noScope: true,
+    ArrowFunctionExpression(path: NodePath<t.ArrowFunctionExpression>) {
+      const {
+        params: [propsParam],
+      } = path.node;
+      // Is the param for props being destructured
+      if (t.isObjectPattern(propsParam)) {
+        path.traverse(appendDestructuredPropsVisitor);
```

Comment:
Instead of traversing again, which is quite inefficient, could we just inspect the first argument and check if it is an object pattern?

Reply:
Hmm in order to get the string source of a node I need to use it's path. Is there a cheaper way to get this?

Reply:
Try `path.get('params')[0].toString()`

Reply:
Ah thanks, I didn't know `get` will return a path too.